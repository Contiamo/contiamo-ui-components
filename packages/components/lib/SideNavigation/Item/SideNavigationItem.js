"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var glamorous_1 = require("glamorous");
var Container = glamorous_1.default.div(function (_a) {
    var theme = _a.theme, active = _a.active;
    var activeBackgroundColor = "rgba(0, 0, 0, 0.2)";
    var hoverBackgroundColor = "rgba(255, 255, 255, 0.07)";
    return {
        position: "relative",
        display: "flex",
        alignItems: "center",
        justifyContent: "flex-start",
        borderRadius: 2,
        width: "fit-content",
        minHeight: 40,
        cursor: "pointer",
        backgroundColor: active ? activeBackgroundColor : "transparent",
        minWidth: "100%",
        whiteSpace: "pre",
        ":hover": {
            backgroundColor: active ? activeBackgroundColor : hoverBackgroundColor
        },
        ":first-child": {
            marginTop: 0
        }
    };
});
var SideNavigationItem = function (props) { return (React.createElement(Container, { key: props.id, css: props.css, className: props.className, active: !!props.active, onClick: props.onClick }, props.children)); };
exports.default = SideNavigationItem;
//# sourceMappingURL=SideNavigationItem.js.map