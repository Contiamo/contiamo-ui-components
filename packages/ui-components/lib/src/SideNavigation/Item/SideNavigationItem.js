"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var glamorous_1 = require("glamorous");
var Container = glamorous_1.default.div(function (_a) {
    var theme = _a.theme, active = _a.active;
    var activeBackgroundColor = "rgba(0, 0, 0, 0.2)";
    var hoverBackgroundColor = "rgba(255, 255, 255, 0.07)";
    return {
        position: "relative",
        display: "flex",
        alignItems: "center",
        justifyContent: "flex-start",
        borderRadius: 2,
        width: "fit-content",
        minHeight: 40,
        cursor: "pointer",
        backgroundColor: active ? activeBackgroundColor : "transparent",
        minWidth: "100%",
        whiteSpace: "pre",
        ":hover": {
            backgroundColor: active ? activeBackgroundColor : hoverBackgroundColor
        },
        ":first-child": {
            marginTop: 0
        }
    };
});
var SideNavigationItem = function (_a) {
    var key = _a.key, className = _a.className, css = _a.css, children = _a.children, onClick = _a.onClick, active = _a.active;
    return (React.createElement(Container, { key: key, css: css, className: className, active: !!active, onClick: onClick }, children));
};
exports.default = SideNavigationItem;
//# sourceMappingURL=SideNavigationItem.js.map