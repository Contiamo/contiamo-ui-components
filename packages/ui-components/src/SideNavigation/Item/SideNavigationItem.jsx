// @flow
import React from "react"
import type { Node } from "react"
import glamorous from "glamorous"

import withTooltip from "../../Tooltip/withTooltip"

type Props = {
  className: string,
  children: Node,
  onClick?: void,
  theme: THEME,
}

const SideNavigationItem = ({ className, children, onClick }: Props): Node =>
    <div className={`${className} SideNavigationItem`} onClick={onClick} role="button" tabIndex="-1">
      {children}
    </div>,
  style = ({ theme }: Props): {} => ({
    position: "relative",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 2,
    cursor: "pointer",

    "&_has-tooltip + &_has-tooltip ": {
      marginTop: theme.spacing ? theme.spacing * 2 : 16
    },

    ":first-child": {
      marginTop: 0,
      marginBottom: theme.spacing ? theme.spacing * 2 : 16
    }
  })

export default glamorous(withTooltip(SideNavigationItem))(style)
