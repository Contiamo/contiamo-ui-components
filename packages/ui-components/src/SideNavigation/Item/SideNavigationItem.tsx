import * as React from "react"
import { SFC } from "react"

import glamorous from "glamorous"

interface Props {
  className?: string
  children: React.ReactNode
  onClick?: () => void
  active?: boolean
  css?: {}
  theme?: Theme
}

const Container = glamorous.div(({ theme, active }: Props): {} => {
  const activeBackgroundColor = "rgba(0, 0, 0, 0.2)"
  return {
    position: "relative",
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-start",
    padding: `${theme.spacing / 2}px ${theme.spacing * 1.7}px`,
    borderRadius: 2,
    width: "fit-content",
    minHeight: 40,
    cursor: "pointer",
    backgroundColor: active ? activeBackgroundColor : "transparent",
    minWidth: "100%",
    whiteSpace: "pre",

    ":hover": {
      backgroundColor: "rgba(255, 255, 255, 0.07)",
    },

    ":first-child": {
      marginTop: 0,
    },
  }
})

const SideNavigationItem: SFC<Props> = ({ className, css, children, onClick, active }: Props) => (
  <Container css={css} className={`${className}`} active={!!active} onClick={onClick} role="button" tabIndex={-1}>
    {children}
  </Container>
)

export default SideNavigationItem
