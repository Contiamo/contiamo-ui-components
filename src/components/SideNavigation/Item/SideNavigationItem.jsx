// @flow
import React, { Component } from 'react';
import glamorous, { Img } from 'glamorous';

type props = {
  className: string,
  children: mixed,
  size?: number,
  theme: THEME,
};

const SideNavigationItem = ({ className, children }: props): React$Element<*> =>
  (<div className={className}>
    {children}
  </div>);

const style = ({ theme, size }: { theme: THEME, size: number }): {} => ({
  position: 'relative',
  width: size || 20,
  height: size || 20,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  borderRadius: 2,
  cursor: 'pointer',

  ':first-child': {
    marginBottom: theme.spacing,
  },

  ':hover > .tooltip': {
    '--offsetLeft': 0,
    opacity: 1,
    pointerEvents: 'all',
  },
});

export default glamorous(SideNavigationItem)(style);
