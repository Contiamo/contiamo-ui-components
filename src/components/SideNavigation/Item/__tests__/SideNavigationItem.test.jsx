// @flow
import React from 'react'
import { shallow } from 'enzyme'

import SideNavigationItem from '../SideNavigationItem'

test('SideNavigationItem component renders', () => {
  const output = shallow(<SideNavigationItem>Hi, I'm an Item</SideNavigationItem>)
  expect(output).toMatchSnapshot()
})

test('SideNavigationItem handles clicks', () => {
  const click = jest.fn()
  const output = shallow(<SideNavigationItem onClick={click}>Hi, I'm an Item</SideNavigationItem>)

  output.simulate('click')
  expect(click).toHaveBeenCalled()
})
