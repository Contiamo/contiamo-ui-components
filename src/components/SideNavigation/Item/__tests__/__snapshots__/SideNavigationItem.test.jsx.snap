// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavigationItem component renders 1`] = `
<div
  className="css-8jobry"
>
  Hi, I'm an Item
</div>
`;
